docs/decisions/0001-null-safety-strategy-for-operaton.md

```diff
@@ -0,0 +1,105 @@
+
+### Confirmation
+
+Compliance will be confirmed via code review and CI checks:
+
+- 100% of public APIs annotated with JSpecify.
+- CI flags nullability violations on public APIs.
+- No externally visible behavior changes (e.g., exception types) in the engine module due to refactors.
+
+## Pros and Cons of the Options
+
+### Status Quo — Keep EnsureUtil/Objects.requireNonNull, no annotations
+
+- Good, because it has no immediate migration cost and preserves current behavior and CI setup.
+- Neutral, because runtime checks can still catch nulls at boundaries but provide no design-time guidance.
+- Bad, because ambiguity about nullability persists; limited tool-assisted safety; diverges from ecosystem direction.
+- Bad, because it adds runtime overhead.
+
+### JSpecify for New Public APIs Only
+
+- Good, because it minimizes churn and adds clarity for newly added surfaces.
+- Good, because runtime null checks can be reduced on hot paths to cut overhead.
+- Good, because null safety is part of the contract/API.
+- Neutral, because mixed annotation coverage can be acceptable temporarily.
+- Bad, because it leaves existing APIs ambiguous for a long time; inconsistent experience for users.
+- Bad, because of inconsistency in null safety across public APIs.
+
+### JSpecify for All Public APIs (Selected)
+
+- Good, because it sets clear, consistent contracts at API boundaries where they matter most; strong tool support; ecosystem alignment.
+- Good, because Spring 7 adopts JSpecify, so projects using Spring and Operaton do not have to deal with multiple null-safety approaches.
+- Good, because it provides consistency in null safety across public APIs.
+- Neutral, because internals can adopt annotations gradually.
+- Bad, because it requires a focused sweep of public APIs and careful handling where exception semantics must be preserved.
+
+### JSpecify Project-Wide
+
+- Good, because it maximizes clarity and static analysis coverage across the codebase.
+- Neutral, because it may be a long-term aspiration after initial adoption.
+- Bad, because it has the highest migration cost and risk of noise/false positives initially.
+
+### Defer Adoption
+
+- Good, because it avoids immediate cost and waits for further maturation.
+- Neutral, because current runtime checks remain in place.
+- Bad, because it delays benefits to users; misses ecosystem alignment; increases future migration burden.
+- Bad, because imitating null safety using `Objects.requireNonNull` creates runtime overhead.
+
+### Use Another Annotation Framework (e.g., JetBrains)
+
+- Good, because it provides similar expressiveness with existing tooling support.
+- Neutral, because many teams are familiar with these annotations.
+- Bad, because it diverges from emerging ecosystem consensus and Spring 7 alignment; offers no clear advantage over JSpecify.
```

Comment:
IMO it is still too early to judge the "ecosystem consensus". Spring's adoption is very important, but JSpecify is still away from being a de-facto standard. This is a bet into the future.